### PR TITLE
UIFC-399: Refactor tests

### PR DESCRIPTION
**Pull request by elsenhans**
https://folio-org.atlassian.net/browse/UIFC-399

### CODE REVIEW

File: src/components/MetadataCollections/MetadataCollections.test.js [hunk 1]
```diff
@@ -1,89 +1,125 @@
...
+    <StripesContext.Provider value={stripes}>
+      <MetadataCollections
+        contentData={data}
+        filterData={tinySources}
+        onNeedMoreData={jest.fn()}
+        queryGetter={jest.fn()}
+        querySetter={jest.fn()}
+        searchString="permitted.yes,selected.yes"
+        selectedRecordId=""
+        visibleColumns={['label', 'mdSource', 'permitted', 'freeContent']}
```
Comment (alb3rtino) [inline src/components/MetadataCollections/MetadataCollections.test.js:27]: `<MetadataCollections>` does not define a `visibleColumns` prop

File: src/components/MetadataSources/MetadataSources.test.js [hunk 2]
```diff
@@ -20,73 +21,113 @@ const filterData = { contacts: [
...
+    <StripesContext.Provider value={stripes}>
+      <MetadataSources
+        contentData={data}
+        filterData={filterData}
+        onNeedMoreData={jest.fn()}
+        queryGetter={jest.fn()}
+        querySetter={jest.fn()}
+        searchString="status.active,status.implementation"
+        selectedRecordId=""
+        visibleColumns={['label', 'sourceId', 'status', 'lastProcessed']}
```
Comment (alb3rtino) [inline src/components/MetadataSources/MetadataSources.test.js:35]: `<MetadataSources>` does not define a `visibleColumns` prop

File: src/index.test.js [hunk 3]
```diff
@@ -109,11 +109,11 @@ const match = {
 const renderWithRouter = (component) => {
   const history = createMemoryHistory();
   return {
-    ...render(withIntlConfiguration(
+    ...withIntlConfiguration(
```
Comment (alb3rtino) [inline src/index.test.js:112]: Please avoid unnecessary spread syntax. You can also use `<MemoryRouter>` for convenience.

File: test/jest/helpers/withIntlConfiguration.js
Comment (alb3rtino) [inline test/jest/helpers/withIntlConfiguration.js:1]: Please rename to `renderWithIntlConfiguration` or similar. `with...` naming scheme should be reserved for HOCs.

File: src/components/MetadataSources/MetadataSources.test.js (hunk 2 above)
Comment (elsenhans) [reply src/components/MetadataSources/MetadataSources.test.js]: Removed `visibleColumns`

File: src/components/MetadataCollections/MetadataCollections.test.js (hunk 1 above)
Comment (elsenhans) [reply src/components/MetadataCollections/MetadataCollections.test.js]: Removed `visibleColumns`

File: src/components/Filters/Filters.test.js [hunk 4]
```diff
@@ -1,63 +1,114 @@
...
         <Filters
-          contentData={filters}
+          contentData={data}
           onNeedMoreData={jest.fn()}
           queryGetter={jest.fn()}
           querySetter={jest.fn()}
           searchString="type.Whitelist,type.Blacklist"
           selectedRecordId=""
-          onChangeIndex={jest.fn()}
+          visibleColumns={['label', 'type']}
```
[the remark this answers is not in the document]
Comment (elsenhans) [reply src/components/Filters/Filters.test.js]: Removed `visibleColumns`

File: src/index.test.js (hunk 3 above)
Comment (elsenhans) [reply src/index.test.js]: Removed spread syntax and use `<MemoryRouter>`

File: test/jest/helpers/withIntlConfiguration.js
Comment (elsenhans) [reply test/jest/helpers/withIntlConfiguration.js:1]: renamed to `renderWithIntlConfiguration`

File: src/index.test.js [hunk 5]
```diff
@@ -106,16 +106,15 @@ const match = {
...
-};
+const history = createMemoryHistory();
+
+const renderWithRouter = (component) => (
+  renderWithIntlConfiguration(
+    <MemoryRouter history={history}>
+      {component}
+    </MemoryRouter>
+  )
+);
```
Comment (alb3rtino) [inline src/index.test.js:117]:
```suggestion

const renderWithRouter = (component) =>
  renderWithIntlConfiguration(<MemoryRouter>{component}</MemoryRouter>);
```

`<MemoryRouter>` takes care of the history.

**Review by alb3rtino — APPROVED** [review]

**Pull request merged**